docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+
+Examples:
+
+.. code-block:: cpp
+
+   /**
+    * Image associated with this map.
+    *
+    * @exceptsafe Shall not throw exceptions.
+    */
+   ImageF getImage() const noexcept;
+
+.. code-block:: cpp
+
+   /**
+    * Apply a user-specified transformation to an image.
+    *
+    * @exceptsafe If `transform` does not leak resources or corrupt data in the
+    *     event of an exception, then this method shall leave the program state
+    *     unchanged in the event of an exception. Otherwise, it provides no
+    *     exception safety guarantee.
+    */
+   template <class Func>
+   ImageF transformImage(Func const & transform) const;
+
+.. _cpp-doxygen-related:
+
+Helper Functions
+----------------
+
+Some operations on a class, particularly arithmetic operations, must be implemented as standalone functions even though they are *conceptually* part of a class. These functions should have the ``@relatesalso`` tag, followed by the name of the appropriate class. They will appear on the class's documentation page under the heading "Related Functions". Use this tag sparingly.
+
+For internal consistency, always use ``@relatesalso`` and not the synonymous ``@relatedalso``.
+
+.. _cpp-doxygen-initializer:
+
+Initializer Declaration
+-----------------------
+
+By default, Doxygen shows the values of constants unless they are very long. The ``@showinitializer`` and ``@hideinitializer`` tags override this behavior.
+
+.. _cpp-doxygen-see-also:
+
+See Also
+--------
+
+'See Also' is an optional section used to refer to related code.
+This section can be very useful, but should be used judiciously.
+The goal is to direct users to other functions they may not be aware of, or have easy means of discovering (by looking at the class or package documentation, for example).
+Functions whose documentation further explains parameters used by this function are good candidates.
+
+List each class, function, or method using a ``@see`` tag, followed by a ``@ref`` tag:
+
+.. code-block:: cpp
+
+   /**
+    * Compute an element-wise cosine.
+    *
+    * @see @ref sin
+    * @see @ref tan
+    */
+   vector<double> cos(vector<double> const & angles);
+
+For internal consistency, always use ``@see`` and not the synonymous ``@sa``.
+
+.. _cpp-doxygen-notes:
+
+Notes
+-----
+
+*Notes* is an optional section that provides additional information about the code, possibly including a discussion of the algorithm or known limitations of the code. The notes should be prefixed by a ``@note`` or ``@warning`` command.
+
+For internal consistency, always use ``@note`` and not the synonymous ``@remark`` or ``@remarks``.
+
+This section may include mathematical equations, written in `LaTeX <http://www.latex-project.org/>`_ format:
```

Comment:
Again, the wording was what it was for consistency with the Python documentation. In this case, though, I think it's also an intrinsically good idea -- unless a particular class is highly mathematical in nature, recommending the use of math in the main description may encourage documentation-writers to focus on implementation details rather than an appropriate level of abstraction.